src/App.tsx

```diff
@@ -14,11 +16,114 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const lengthFromServer: number[] = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+const App: React.FC = () => {
+  const [state, setState] = useState({
+    goods: [...goodsFromServer],
+    isVisible: false,
+    length: 1,
+  });
+
+  const reverseGoods = () => {
+    const reverseGood = {
+      ...state,
+      goods: [...state.goods].reverse(),
+    };
+
+    setState(reverseGood);
```

Comment:
you need the current state, so you can use callback on setState
https://reactjs.org/docs/hooks-reference.html#usestate

```suggestion
    setState((currentState) => and here return reversed goods);
```

Reply:
I reading this tomorrow)